src/docfx/lib/markdown/validation/DocsValidationExtension.cs

```diff
@@ -231,5 +240,126 @@ static string GetContainerInlineContent(ContainerInline containerInline)
                 return content.ToString();
             }
         }
+
+        private static void BuildTableNodes(MarkdownObject node, MarkdownEngine markdownEngine, List<TableNode> tableNodes, bool isCanonicalVersion)
+        {
+            TableNode? tableNode = null;
+            switch (node)
+            {
+                case Table table:
+                    var parsedNode = TryParseTable(table);
+                    tableNode = CreateValidationNode<TableNode>(isCanonicalVersion, node) with
+                    {
+                        RowHeaders = parsedNode.RowHeaders,
+                        ColumnHeaders = parsedNode.ColumnHeaders,
+                        IsSuccessParsed = parsedNode.IsSuccessParsed,
+                    };
+                    break;
+                case ParagraphBlock:
+                    if (TryDetectTable((ParagraphBlock)node, markdownEngine))
+                    {
+                        tableNode = CreateValidationNode<TableNode>(isCanonicalVersion, node) with { IsSuccessParsed = false };
+                    }
+                    break;
+                default:
+                    break;
+            }
+            if (tableNode != null)
+            {
+                tableNodes.Add(tableNode);
+            }
+        }
+
+        private static TableNode TryParseTable(Table table)
+        {
+            if (table.Count == 0)
+            {
+                return new TableNode { IsSuccessParsed = false, };
+            }
+            var columnHeaders = new List<TableCellNode>();
+            var rowHeaders = new List<TableCellNode>();
+            var columnHeaderRow = (TableRow)table.First();
+            foreach (TableCell cell in columnHeaderRow)
+            {
+                columnHeaders.Add(ParseCell(cell));
+            }
+            foreach (TableRow row in table)
+            {
+                if (row.Count == 0)
+                {
+                    rowHeaders.Add(new TableCellNode { IsVisible = false });
+                }
+                else
+                {
+                    rowHeaders.Add(ParseCell((TableCell)row.First()));
+                }
+            }
+            return new TableNode
+            {
+                ColumnHeaders = columnHeaders.ToArray(),
+                RowHeaders = rowHeaders.ToArray(),
+                IsSuccessParsed = true,
+            };
+        }
+
+        private static TableCellNode ParseCell(TableCell cell)
+        {
+            if (cell == null || cell.Count == 0)
+            {
+                return new TableCellNode
+                {
+                    IsVisible = false,
+                    IsEmphasis = false,
+                };
+            }
+            var block = cell.LastChild;
```

Comment:
why use the Last Child? consider below case:
```md
| asf | fda | 
| -- | -- |
| sdafasd ![sdf](asdfdasf) **sadffd** sds | asdfdsf |

```

Reply:
We could say in order to be an emphasis, aside from whitespaces, the block MUST be a paragraph block that starts and ends with `EmphasisInline`.

Is `*a*` also considered valid or we only allow `**a**`? `EmphasisInline.DelimiterCount` tells the difference.